src/poliastro/tests/tests_twobody/test_perturbations.py

```diff
@@ -101,3 +110,66 @@ def accel(t0, state, k):
     final = initial.propagate(initial.period, method=cowell, ad=accel)
     assert_quantity_allclose(final.r, initial.r)
     assert_quantity_allclose(final.v, initial.v)
+
+
+moon_heo = {'body': Moon, 'tof': 60, 'raan': -0.06 * u.deg, 'argp': 0.15 * u.deg, 'inc': 0.08 * u.deg,
+            'orbit': [26553.4 * u.km, 0.741 * u.one, 63.4 * u.deg, 0.0 * u.deg, -10.12921 * u.deg, 0.0 * u.rad],
+            'period': 28}
+
+moon_leo = {'body': Moon, 'tof': 60, 'raan': -2.18 * 1e-4 * u.deg,
+            'argp': 15.0 * 1e-3 * u.deg, 'inc': 6.0 * 1e-4 * u.deg,
+            'orbit': [6678.126 * u.km, 0.01 * u.one, 28.5 * u.deg, 0.0 * u.deg, 0.0 * u.deg, 0.0 * u.rad],
+            'period': 28}
+
+moon_geo = {'body': Moon, 'tof': 60, 'raan': 6.0 * u.deg, 'argp': -11.0 * u.deg, 'inc': 6.5 * 1e-3 * u.deg,
+            'orbit': [42164.0 * u.km, 0.0001 * u.one, 1 * u.deg, 0.0 * u.deg, 0.0 * u.deg, 0.0 * u.rad],
+            'period': 28}
+
+sun_heo = {'body': Sun, 'tof': 720, 'raan': -0.31 * u.deg, 'argp': 0.84 * u.deg, 'inc': 0.23 * u.deg,
+           'orbit': [26553.4 * u.km, 0.741 * u.one, 63.4 * u.deg, 0.0 * u.deg, -10.12921 * u.deg, 0.0 * u.rad],
+           'period': 365}
+
+sun_leo = {'body': Sun, 'tof': 720, 'raan': -17.0 * 1e-3 * u.deg, 'argp': 0.11 * u.deg, 'inc': -0.3 * 1e-4 * u.deg,
+           'orbit': [6678.126 * u.km, 0.01 * u.one, 28.5 * u.deg, 0.0 * u.deg, 0.0 * u.deg, 0.0 * u.rad],
+           'period': 365}
+
+sun_geo = {'body': Sun, 'tof': 720, 'raan': 25.0 * u.deg, 'argp': -22 * u.deg, 'inc': 0.125 * u.deg,
+           'orbit': [42164.0 * u.km, 0.0001 * u.one, 1 * u.deg, 0.0 * u.deg, 0.0 * u.deg, 0.0 * u.rad],
+           'period': 365}
+
+
+@pytest.mark.parametrize('test_params', [
+    moon_heo, moon_geo, moon_leo,
+    sun_heo, sun_geo,
+    pytest.param(sun_leo, marks=pytest.mark.skip(reason="here agreement required rtol=1e-10, too long for 720 days"))
+])
+def test_3rd_body_Curtis(test_params):
+    # based on example 12.11 from Howard Curtis
+    body = test_params['body']
+
+    j_date = 2454283.0
+    tof = (test_params['tof'] * u.day).to(u.s).value
+    body_r = build_ephem_interpolant(body, test_params['period'], (j_date, j_date + test_params['tof']), rtol=1e-2)
+
+    epoch = Time(j_date, format='jd', scale='tdb')
+    initial = Orbit.from_classical(Earth, *test_params['orbit'], epoch=epoch)
+    r, v = cowell(initial, np.linspace(0, tof, 400), rtol=1e-10, ad=third_body,
+                  k_third=body.k.to(u.km**3 / u.s**2).value, third_body=body_r)
+
+    incs, raans, argps = [], [], []
+    for ti, ri, vi in zip(np.linspace(0, tof, 400), r, v):
+        _, _, inc, raan, argp, _ = rv2coe(Earth.k.to(u.km**3 / u.s**2).value, ri, vi)
+
+        # fighting %2pi
```

Comment:
I guess this is done to put argp, raan and inc in the range [-pi, pi]? In that case, you could use http://docs.astropy.org/en/stable/api/astropy.coordinates.Angle.html

```
In [45]: a = Angle([-20.0, 150.0, 350.0] * u.deg)

In [46]: a.wrap_at(180 * u.deg)
Out[46]: <Angle [-20., 150., -10.] deg>

In [47]: import numpy as np

In [48]: a.wrap_at(np.pi * u.rad)
Out[48]: <Angle [-20., 150., -10.] deg>
```
```